src/resolve/resolved-dns-packet.c

```diff
@@ -1672,6 +1697,38 @@ static bool loc_size_ok(uint8_t size) {
         return m <= 9 && e <= 9 && (m > 0 || e == 0);
 }
 
+static bool dns_svc_param_is_valid(DnsSvcParam *i) {
```

Comment:
we usually start out validity check functions with a check if the specified parameter is NULL or not.

i.e. add `if (!i) return false` here as first check?